bciers/apps/dashboard/middlewares/withAuthorizationDashboard.ts

```diff
@@ -7,39 +7,24 @@ import {
 
 import { MiddlewareFactory } from "@bciers/middlewares";
 import { getToken } from "@bciers/actions";
+import { FrontEndRoles } from "@bciers/utils/enums";
+import isInAllowedPath from "@bciers/utils/isInAllowedList";
 /*
 Access control logic is managed using Next.js middleware and NextAuth.js authentication JWT session.
 The middleware intercepts requests, and for restricted areas...
 Checks for a valid user session, and extracts user information from the JWT session.
 Based on JWT properties of identity_provider and role, the middleware dynamically rewrites the request URL
 to the appropriate folder structure.
  */
-
-const onboarding = "onboarding";
-const dashboard = "dashboard";
-// Function to check if the path is in the unauthenticated allow list
-const isUnauthenticatedAllowListedPath = (pathname: string): boolean => {
-  const authList = ["auth", "unauth", "demo"];
-  return authList.some((path) => pathname.includes(path));
+const paths = {
+  auth: "auth",
+  unauth: "unauth",
+  onboarding: "onboarding",
+  dashboard: "dashboard",
+  profile: "profile",
 };
-
-// Function to check if the path is in the authenticated allow list
-// const isAuthenticatedAllowListedPath = (pathname: string) => {
-//   const allowList = [dashboard, "registration", "reporting"];
-//
-//   // Split the pathname into segments
-//   const segments = pathname.split("/");
-//
-//   // Iterate through each segment
-//   for (const segment of segments) {
-//     // Check if any value in the allowList matches the current segment
-//     if (allowList.includes(segment)) {
-//       return true; // Return true if match found
-//     }
-//   }
-//
-//   return false; // Return false if no match found
-// };
+export const authAllowedPaths = [paths.dashboard, paths.profile];
+const unauthAllowedPaths = [paths.auth, paths.auth];
```

Comment:
Is this line right? The variable name suggests something about UNauth, and both array items are the same

Reply:
@BCerki you're the best